### ISSUE TITLE
Final live training panel resets the last loss to 0.0000

### ISSUE THREAD

**Issue by Amix29** [MakazhanAlpamys/Soup#541]
## Environment

- Soup 0.73.3
- Transformers 5.12.1
- Apple M4 Max / MPS

## Observed behavior

After a successful 128-step SFT run, the final live training panel displayed:

```text
Loss:  0.0000    LR: 0.00e+00
```

The Trainer history and Soup's completion panel both contained the real values:

```text
last step loss: 1.3589
completion panel: Loss: 4.3004 -> 1.3589
```

The final Trainer summary log contains runtime/throughput fields but no per-step `loss`. `SoupTrainerCallback.on_log()` currently defaults a missing loss to `0.0` and overwrites the last valid value in `TrainingDisplay`.

## Expected behavior

Summary-only log events should preserve the most recent training loss, learning rate and gradient norm in the live panel rather than replacing missing values with zero. A genuine logged `0.0` must still be displayed as zero.

I reproduced this in the Qwen3.8-27B Apple Silicon smoke test and am preparing a focused callback regression test and fix.

**Comment by MakazhanAlpamys** [comment]
Closed by #544 from @Amix29. The live training panel no longer replaces final loss/LR/grad norm with synthetic zeroes from summary-only logs; the cache-assignment mutation is caught by a named regression.
